src/cdk/overlay/overlay-container.ts

```diff
@@ -19,27 +19,29 @@ export class OverlayContainer implements OnDestroy {
 
   private _themeClass: string;
 
+  ngOnDestroy() {
+    if (this._containerElement && this._containerElement.parentNode) {
+      this._containerElement.parentNode.removeChild(this._containerElement);
+    }
+  }
+
   /**
-   * Base theme to be applied to all overlay-based components.
+   * Adds a CSS class to the overlay container.
+   * @param cssClass The CSS class to add.
    */
-  get themeClass(): string { return this._themeClass; }
-  set themeClass(value: string) {
-    if (this._containerElement) {
-      if (this._themeClass) {
-        this._containerElement.classList.remove(this._themeClass);
-      }
-
-      if (value) {
-        this._containerElement.classList.add(value);
-      }
+  addClass(cssClass: string) {
```

Comment:
Since the `getContainerElement()` is public, I'm not sure that these methods are even necessary.

Reply:
Yeah, I considered that. I was worried that the change would be "too complicated" for people, but maybe I'm overthinking it. I'll just remove the `themeClass` completely.

Reply:
IMO we still need some easy way for people to pass the theme to overlays, though. Perhaps a provider that injects the overlay container and applies the class?

Reply:
Is `getContainerElement().classList.add('dark-theme')` that must worse than `themeClasss = 'dark-theme';`?

Reply:
I don't think it is, but I guess that it's not as obvious that that's the way to theme all components.

Reply:
IMO while `getContainerElement().classList.add('dark-theme')` isn't immediately obvious, it reduces the magic, which is nice.

Just don't forget [the example in the theming guide](https://github.com/angular/material2/blob/master/guides/theming.md#multiple-themes-and-overlay-based-components) needs to be updated with whichever solution!